src/main/java/org/spring/FullVersion/CollectionFamilyDao.java

```diff
@@ -48,6 +49,27 @@ public void saveFamily(Family family) {
 
     }
 
+    @Override
+    public void saveData(List<Family> families) {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream("families.dat"))) {
```

Comment:
You can create a variable FILE_PATH, and set it families.dat, and use it instead write it every time